timemachine/cpp/src/kernels/k_nonbonded_common.cuh

```diff
@@ -12,23 +12,155 @@ static const int NONBONDED_KERNEL_THREADS_PER_BLOCK = 256;
 #define PI 3.141592653589793115997963468544185161
 #define TWO_OVER_SQRT_PI 1.128379167095512595889238330988549829708
 
-double __device__ __forceinline__ real_es_factor(double real_beta, double dij, double inv_d2ij, double &erfc_beta_dij) {
-    double beta_dij = real_beta * dij;
+// f64 switch fxn
+double __device__ __forceinline__ switch_fn(double dij) {
+    // constants
+    constexpr double cutoff = 1.2;
+    constexpr double inv_cutoff = 1 / cutoff;
+    constexpr double pi = static_cast<double>(PI);
+
+    // (dij/cutoff)^8
+    double dij_k = dij * inv_cutoff;
+    double dij_k2 = dij_k * dij_k;
+    double dij_k4 = dij_k2 * dij_k2;
+    double dij_k8 = dij_k4 * dij_k4;
+
+    // cos(0.5*pi*(dij/cutoff)^8)^3
+    double cos_arg = cos(0.5 * pi * dij_k8);
+    double cos_arg3 = cos_arg * cos_arg * cos_arg;
+    return cos_arg3;
+}
+
+double __device__ __forceinline__ d_switch_fn_dr(double dij) {
+    // constants
+    constexpr double cutoff = 1.2;
+    constexpr double pi = static_cast<double>(PI);
+
+    // cutoff^-8
+    double inv_cutoff = 1.0 / cutoff;
```

Comment:
nit: though likely to disappear if you allow swaping the cutoff, use constexpr here

Reply:
(and also for `k2`, `k4`, `k8`?)

Reply:
Made these constexpr in https://github.com/proteneer/timemachine/pull/1290/commits/92c16263e3706cde4244e136370fb16bcdc040fb